### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by HorskayaAnna**
Task 2. Horskaya Anna

### CODE REVIEW

File: DEV-2/DEV-2/GetStringWithEvenSymbols.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            processedString = arg;
+        }
+
+        /// <summary>
+        /// This metod Delete Odd Symbols in string
+        /// </summary>
+        public void DeleteOddSymbols()
+        {
+            processedString = processedString.Replace(" ", "");
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/GetStringWithEvenSymbols.cs]: Why? this is bad idea to remove spaces, bug

File: DEV-2/DEV-2/GetStringWithEvenSymbols.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            StringBuilder bufferString = new StringBuilder(10);
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/GetStringWithEvenSymbols.cs]: Why 10?

File: DEV-2/DEV-2/GetStringWithEvenSymbols.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            int position=0;
+            for (int i = 1; i < processedString.Length;i+=2)
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/GetStringWithEvenSymbols.cs]: Why do you use two StringBuilders?

File: DEV-2/DEV-2/EntryPoint.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Input string, please");
+            StringBuilder inputString = new StringBuilder(Console.ReadLine());
+            if (inputString.Length > 0)
+            {
+                GetStringWithEvenSymbols delete = new GetStringWithEvenSymbols(inputString);
+                delete.DeleteOddSymbols();
+            }
+            else Console.WriteLine("Error:Empty string");
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/EntryPoint.cs]: Code style

File: DEV-2/DEV-2/GetStringWithEvenSymbols.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            {
+                bufferString.Insert(position, processedString[i]);
+                position++;
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/GetStringWithEvenSymbols.cs]: What for position? looks like very complicated, simplify

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]

File: DEV-2/DEV-2/EntryPoint.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+{
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Input string, please");
+            StringBuilder inputString = new StringBuilder(Console.ReadLine());
+            if (inputString.Length > 0)
+            {
+                GetStringWithEvenSymbols delete = new GetStringWithEvenSymbols(inputString);
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/EntryPoint.cs]: bad variable name

File: DEV-2/DEV-2/GetStringWithEvenSymbols.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Text;
+
+namespace DevTask2
+{
+    /// <summary>
+    /// This class contains method, which delete odd symbols 
+    /// in string and output this string
+    /// </summary>
+    class GetStringWithEvenSymbols
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/GetStringWithEvenSymbols.cs]: Class name should be noun

File: DEV-2/DEV-2/GetStringWithEvenSymbols.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            processedString = arg;
+        }
+
+        /// <summary>
+        /// This metod create new string with only even index symbols
+        /// </summary>
+        public void DeleteOddSymbols()
+        {
+            string bufferString = "";
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/GetStringWithEvenSymbols.cs]: use string builder instead operation with immutable type

File: DEV-2/DEV-2/GetStringWithEvenSymbols.cs [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+            for (int i = 0; i < processedString.Length; i+=2)
+            {
+                bufferString += processedString[i];
+            }
+            Console.WriteLine($"String with even symbols : {bufferString}");
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/GetStringWithEvenSymbols.cs]: single responsibility violation

File: DEV-2/DEV-2/GetStringWithEvenSymbols.cs [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+            processedString = arg;
+        }
+
+        /// <summary>
+        /// This metod create new string with only even index symbols
+        /// </summary>
+        public void DeleteOddSymbols()
+        {
+            string bufferString = "";
+            for (int i = 0; i < processedString.Length; i+=2)
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/GetStringWithEvenSymbols.cs]: codestyle violation

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: DEV-2/DEV-2/EntryPoint.cs [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+{
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Input string, please");
+            StringBuilder inputString = new StringBuilder(Console.ReadLine());
+            if (inputString.Length > 0)
+            {
+                GetterStringWithEvenSymbols evener = new GetterStringWithEvenSymbols(inputString);
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/EntryPoint.cs:14]: evener? is it English?

File: DEV-2/DEV-2/GetStringWithEvenSymbols.cs [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+        /// <param name="arg">
+        /// string taken from the console
+        /// </param>
+        public GetterStringWithEvenSymbols(StringBuilder arg)
+        {
+            processedString = arg;
+        }
+
+        /// <summary>
+        /// This metod create new string with only even index symbols
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/GetStringWithEvenSymbols.cs:26]: wrong description

File: DEV-2/DEV-2/GetStringWithEvenSymbols.cs [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+        /// </summary>
+        public void StringWithEvenSymbolsCreator()
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/GetStringWithEvenSymbols.cs:28]: wrong name

File: DEV-2/DEV-2/GetStringWithEvenSymbols.cs [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            Console.WriteLine($"String with even symbols : {GetEvenPositionsStringBuilder()}");
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/GetStringWithEvenSymbols.cs:30]: single responsibility violation

**Review by AlexanderSoroka — APPROVED** [review]

**Pull request merged**